src/screens/SignUpScreen/index.tsx

```diff
@@ -344,6 +365,7 @@ class SignUp extends React.Component<Props> {
     private handleSignUp = () => {
         const { configs, i18n, captcha_response } = this.props;
         const {
+            nickname,
             email,
             password,
             refId,
```

Comment:
This function looks enormous now, could you please rewrite it
For example
const payload = {
   email,
   password,
   refid: refId,
   data: JSON.stringify({
     language: i18n,
  });

Switch captcha_type
...
payload = { ...payload, captcha response}

if (nicknameEnabled) {
    payload = { ...payload, nickname }
}